modules/configuring-private-storage-endpoint-azure-vnet-subnet-iro-discovery.adoc

```diff
@@ -0,0 +1,133 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/configuring-private-cluster.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="configuring-private-storage-endpoint-azure-vnet-subnet-iro-discovery"]
+= Setting up a private storage endpoint on Microsoft Azure by configuring the Image Registry Operator to discover VNet and subnet names
+
+The following procedure shows you how to set up a private storage endpoint on Microsoft Azure by configuring the Image Registry Operator to discover VNet and subnet names.
+
+.Prerequisites 
+
+* You have configured an image registry to run on Azure.
+* Your network has been set up using the Installer Provisioned Infrastructure (IPI) installation method. 
++
+For users with VNet and subnet names, see "Configuring a private storage endpoint on Azure with user-provided VNet and subnet names". 
+
+.Procedure 
+
+. Edit the Image Registry Operator `config` object and omit your VNet and subnet names:
++
+[source,terminal]
+----
+$ oc edit configs.imageregistry/cluster
+----
++
+[source,terminal]
+----
+# ...
+spec:
+  [...]
+   storage:
+      azure:
+        [...]
+        networkAccess:
+          type: Internal
+# ...
+----
+
+. Enter the following command to confirm that the Operator has completed provisioning. This might take a few minutes. 
++
+[source,terminal]
+----
+$ oc get configs.imageregistry/cluster -o=jsonpath="{.spec.storage.azure.privateEndpointName}" -w
+----
+
+. Optional. If the registry is exposed by a route, and you are configuring your storage account to be private, you must disable redirect if you want pulls external to the cluster to continue to work. Enter the following command to disable redirect on the Image Operator configuration:
++
+[source,terminal]
+----
+$ oc patch configs.imageregistry cluster --type=merge -p '{"spec":{"disableRedirect": true}}'
+----
++
+[NOTE]
+====
+When redirect is enabled,  pulling images from outside of the cluster is not expected to work.
+====
+
+.Verification
+
+Use the following verification steps to ensure that users within the same cluster can pull images from the registry.
```

Comment:
Both verification steps (`Verification` and `Verification with redirect enabled`) expect the registry to have an exposed route. Having an exposed route is not (and should not be) a requirement for enabling private endpoints - it's confusing that the procedure doesn't require exposing a route but the verification does... We need to either change the verification steps so that it pulls an image from a pod within the cluster (which is what this sentence claims this verification steps do) or we explain that the verification expects the registry to be exposed via the default route, i.e `defaultRoute: true`.
I prefer the former.

Reply:
@flavianmissi Happy to make it the former, but not sure if I know what that entails. I'm taking a stab at it: 

```suggestion

. $ REGISTRY=$(oc get svc -n openshift-image-registry --template='{{ .spec.clusterIP }}')

. $ podman login --tls-verify=false -u unused -p $(oc whoami -t) ${REGISTRY}

. $ podman pull --tls-verify=false ${REGISTRY}/openshift/tools

```

Sorry & Thanks!

Reply:
You have to run the command from within the cluster. In other words, your podman client needs to be inside the cluster network.
I'm not sure what's the best way to tell customers to do it, here are a couple of options:
* import an image via `oc image-import` then create an app with it (this will use the internal route by default)
* use `oc debug node/<node-name>` to login into a node, run the suggested chroot command in the prompt then pull an image using the **internal** registry route. 
What do you think? The node debug is the closest to what you're currently documenting.